### PR TITLE
docs

### PR DESCRIPTION
**Pull request by DanielVoogsgerd**
This should resolve all the warnings in the docs.
Furthermore, this should from now on disallow the introductions of warnings in CI

### CODE REVIEW

File: brane-cli/src/errors.rs [hunk 1]
```diff
@@ -1818,7 +1818,7 @@ impl Error for UtilError {}
 /// Defines errors that relate to finding our directories.
 #[derive(Debug)]
 pub enum DirError {
-    /// Failed to find a user directory. The `what` hints at the kind of user directory (fill in "<WHAT> directory", e.g., "config", "data", ...)
+    /// Failed to find a user directory. The `what` hints at the kind of user directory (fill in "`<WHAT>` directory", e.g., "config", "data", ...)
```
Comment (Lut99) [inline brane-cli/src/errors.rs:1821]: I guess I don't agree with clippy here xD the "what" is part of the string, shouldn't contain any backticks in it. If you like, I'd prefer then to have the whole string backtick'ed

(I'm expressing that <WHAT> is a placeholder in the string to be replaced with the examples "config", "data", etc)

File: brane-cli/src/run.rs [hunk 2]
```diff
@@ -686,7 +686,7 @@ pub async fn initialize_instance_vm(
 ///
 /// # Arguments
 /// - `state`: The DummyVmState that we use to run the dummy VM.
-/// - `what`: The thing we're running. Either a filename, or something like '<stdin>'.
+/// - `what`: The thing we're running. Either a filename, or something like `<stdin>`.
```
Comment (Lut99) [inline brane-cli/src/run.rs:689]: Can also remove the `<` and `>` here

File: brane-ctl/src/errors.rs [hunk 3]
```diff
@@ -462,7 +462,7 @@ pub enum PackagesError {
     NodeConfigLoadError { err: brane_cfg::info::YamlError },
     /// The given node type is not supported for this operation.
     ///
-    /// The `what` should fill in the `<WHAT>` in: "Cannot <WHAT> on a ... node"
+    /// The `what` should fill in the `<WHAT>` in: "Cannot `<WHAT>` on a ... node"
```
Comment (Lut99) [inline brane-ctl/src/errors.rs:465]: Same as above; the quote doesn't include backticks. Quote the entire string?

File: brane-exe/src/pc.rs [hunk 4]
```diff
@@ -111,7 +111,7 @@ impl ProgramCounter {
 
     /// Returns a ProgramCounter that points to the given edge within the same function.
     ///
-    /// This function returns a new instance. To update an existing one, use [`Self::jump_mut`](jump_mut).
+    /// This function returns a new instance. To update an existing one, use [`ProgramCounter::jump_mut()`].
```
Comment (Lut99) [inline brane-exe/src/pc.rs:114]: OK yeah thanks, I've been wildly inconsistent in this. I'll use that from now on xD

**Review by Lut99 — CHANGES_REQUESTED** [review]
Thanks! I hadn't thought about rustdoc warnings. I've left a few minor comments, though, for maximum matching my intensions :)

File: brane-cli/src/errors.rs (hunk 1 above)
Comment (DanielVoogsgerd) [reply brane-cli/src/errors.rs:1821]: The main problem is that it was trying to render the `<WHAT>` as HTML. Putting it in backticks works around that issue, but I think I can also just escape the `<` and `>`. I will try that.

**Comment by DanielVoogsgerd** [comment]
Comments should be resolved.

**Comment by Lut99** [comment]
They are. Want to rebase #163 first?

**Review by Lut99 — APPROVED** [review]

**Pull request merged**